src/readme.js

```diff
@@ -0,0 +1,199 @@
+
+`;
+  }
+  if (!installationIncluded && testIncluded) {
+    return `
+
+## Table of contents
+1. [Description](#description)
+2. [User Story](#user-story)
+3. [User Guidance](#guidance)
+    1. [Usage](#usage)
+    1. [Test Instructions](#test)
+    2. [Contribution Guidelines](#contribution)
+4. [Questions](#questions)
+
+`;
+  } else {
+    return `
+
+## Table of contents
+1. [Description](#description)
+2. [User Story](#user-story)
+    1. [Usage](#usage)
+    3. [User Guidance](#guidance)
+    1. [Contribution Guidelines](#contribution)
+4. [Questions](#questions)
+
+`;
+  }
+};
+
+const buildDescription = ({ description, userStory }) => `
+<a id="description"></a>
+### Description of application 
+${description}
+
+<a id="user-story"></a>
+### User Story 
+
+${userStory}
+`;
+
+const buildInstructionsList = (instructionsArray) => {
+  const list = instructionsArray.join("\n -");
+  return ` -${list}`;
+};
+
+const buildGuidance = (
+  { installationIncluded, testIncluded, usage, contribution },
+  instructionsArray,
+  testArray
+) => {
+  const instruct = buildInstructionsList(instructionsArray);
+  const test = buildInstructionsList(testArray);
+
+  if (installationIncluded && testIncluded) {
+    return `
+<a id="guidance"></a>
+### User guidance
+
+<a id="usage"></a>
+#### App Usage
+
+    ${usage}
+
+<a id="installation"></a>
+#### Installation Instructions 
+
+\`\`\`    
+${instruct}
+\`\`\`
+
+<a id="test"></a>
+#### Test Instructions 
+   
+\`\`\`    
+${test}
+\`\`\`
+    
+<a id="contribution"></a>
+#### Contribution Guidelines 
+    
+${contribution}
+`;
+  }
+  if (installationIncluded && !testIncluded) {
+    return `
+<a id="guidance"></a>
+### User guidance
+
+<a id="usage"></a>
+#### App Usage
+
+  ${usage}
+
+<a id="installation"></a>
+#### Installation Instructions 
+
+\`\`\`    
+${instruct}
+\`\`\`
+
+<a id="contribution"></a>
+#### Contribution Guidelines 
+    
+    ${contribution}`;
+  }
+  if (!installationIncluded && testIncluded) {
+    return `<a id="guidance"></a>
+### User guidance
+
+<a id="usage"></a>
+#### App Usage
+
+    ${usage}
+
+<a id="test"></a>
+#### Test Instructions 
+    
+\`\`\`    
+${test}
+\`\`\`
+
+<a id="contribution"></a>
+#### Contribution Guidelines 
+    
+    ${contribution}`;
+  } else {
+    return `
+<a id="guidance"></a>
+### User guidance
+
+<a id="usage"></a>
+#### App Usage
+
+    ${usage}
+
+
+<a id="contribution"></a>
+#### Contribution Guidelines 
+    
+    ${contribution}
+    `;
+  }
+};
+
+const buildQuestions = ({ email, githubName }) => `
+<a id="questions"></a>
+### Questions 
+Should you have any questions regarding this project please reach me via email or via my GitHub using the details below:
+
+-Email address: ${email}
+
+-GitHub username : <a href='https://github.com/${githubName}'>${githubName}<a>`;
+
+// write README - include readme markup
+
+const writeReadme = ({
+  readmeAnswers: answers,
+  instructionsArray,
+  testArray,
+}) =>
```

Comment:
Please avoid using string concatenation like this, but instead utilize the template string syntax:

```
`
${ buildTitleAndBadge(answers) }
${ buildTableOfContents(answers) }
${ buildDescription(answers) }
${ buildGuidance(answers, instructionsArray, testArray) }
${ buildQuestions(answers) };
`

```